sway-core/src/type_system/type_info.rs

```diff
@@ -1093,6 +1233,35 @@ impl TypeInfo {
         }
     }
 
+    pub(crate) fn can_change(&self) -> bool {
+        // TODO: there might be an optimization here that if the type params hold
+        // only non-dynamic types, then it doesn't matter that there are type params
```

Comment:
by non-dynamic do you mean only resolved types? i.e. no generics?

Reply:
Yes! But also, I didn't know if I should call it that directly because the "can_change" method also checks for `TypeInfo::Contract` and such. What do you think?

Reply:
Hm I think the term dynamic type invokes thoughts of scripting languages and dynamic typing, so I was confused by the comment initially. What about  "if the type params only hold concrete types"?

Reply:
I like it!